### PR TITLE
oop-special_class_method-medium-q14

### PR DESCRIPTION
**Pull request by KaiKhademi**
## PR Checklist
- [x]  Are helper methods really needed here?
    > Opt for hardcoding input/output into the input provider when possible
- [x]  Are the assertion fail messages good?
    - [x]  Logical (does it make sense?)
    - [x]  Readable (can a student understand what’s wrong?)
    - [x]  Not too feed-y (does it give away the answer?)
    - [x]  Grammatically sound (is the grammar and punctuation in the sentence proper?)
    - [x]  Specific (does the fail message relate to the question?)
- [x]  Is the code formatted well?
    > Run `CTRL` + `ALT` + `L` in IntelliJ
- [x]  EOF newlines
- [x]  Do the comments for Parsons follow the correct convention?
- [x]  Is the question being tested sufficiently?
    - [x]  Are there representative cases to test the program?
    - [x]  What about edge cases? (empty strings, negatives, infinity, off-by-one errors, etc)
        > If the question doesn’t specify what to do, mark it as a bad question
- [x]  Are the test method names good?
    > Test method names show up in a “What went well” section on the website, so “mathTest” will become “What went well: math test”, which doesn’t make much sense. A better name in this example would be “calculatesTaxCorrectly” so it shows up as “What went well: calculates tax correctly”

### CODE REVIEW

File: src/oop/special_class_method/medium/q14/QuestionText.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+Your `Water` class already has the following items:
+
+- Attributes:
+    - A `String` called **quality**
+    - A `double` called **volume**
+    - These are not visible from other classes
+
+- Constructor:
+    - a `Water` constructor that is visible to other classes
+      initialize the **quality** and **volume** attributes to `Good` and 1000.0.
```
Comment (SamamthaH) [inline src/oop/special_class_method/medium/q14/QuestionText.md:19]: fix line formatting

File: src/oop/special_class_method/medium/q14/MainTest.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    private final String initialValue1 = "Good";
+    private final double initialValue2 = 1000.0;
+    private ObjectTest testClass;
+
+    @BeforeEach
+    public void setUp() throws Throwable {
+        String classString = "oop.special_class_method.medium.q14." + className;
+        this.testClass = new ObjectTest(classString);
+        String modifiedClassMessage =
+                "You have modified the provided portions of class " + className + ". Please revert them to the original state.";
```
Comment (SamamthaH) [inline src/oop/special_class_method/medium/q14/MainTest.java:27]: This should probably be clearer on what has been modified

**Review by SamamthaH — CHANGES_REQUESTED** [review]

File: src/oop/special_class_method/medium/q14/MainTest.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+                "You have modified the provided portions of class " + className + " constructor. Please revert them to the original state.");
+
+    }
+
+    @RepeatedTest(10)
+    public void correctGetVolumeMethod() throws Throwable {
+        Object testInstance = testClass.createInstance();
+        double value = Math.random() * 10000;
+        testClass.setFieldValue(testInstance, value, attributeName2);
+        Object getMethodOutput = testClass.callMethod(getAttributeMethodName2, testInstance);
```
Comment (SamamthaH) [inline src/oop/special_class_method/medium/q14/MainTest.java:49]: use .hasMethod first

File: src/oop/special_class_method/medium/q14/MainTest.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        Object testInstance = testClass.createInstance();
+        double updatedValue = Math.random() * 10000;
+        Object[][] setMethodArguments = {
+                {updatedValue, double.class}
+        };
+        Class<?>[] methodModifierClasses = {
+                double.class
+        };
+        assertTrue(testClass.hasMethod(setAttributeMethodName2, methodModifierClasses, Void.TYPE),
+                "Your " + setAttributeMethodName2 + " method does not have the correct name, return type, or parameters.");
```
Comment (SamamthaH) [inline src/oop/special_class_method/medium/q14/MainTest.java:67]: This assert should be split in two so return type can be its own assert

File: src/oop/special_class_method/medium/q14/QuestionText.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
+# oop-special_class_method-medium-q14
+
+## Question Text
+
+As an employee of a material processing facility, you are tasked with modelling the facility's functionalities. Make a
+Java class called **Water** that has an attribute called **quality** and another attribute called **volume**.
```
Comment (SamamthaH) [inline src/oop/special_class_method/medium/q14/QuestionText.md:6]: get rid of duplicate class details

**Review by SamamthaH — CHANGES_REQUESTED** [review]

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]
See my feedback on the other questions in this category and apply it to this PR.

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]
Please look at the other questions Sam and I have given feedback on in this category and apply it to this question.

**Review by SamamthaH — APPROVED** [review]

**Pull request merged**